hedera-node/hedera-app/src/main/java/com/hedera/node/app/signature/impl/SignatureVerifierImpl.java

```diff
@@ -68,14 +81,13 @@ public Map<Key, SignatureVerificationFuture> verify(
             final var kind = sigPair.sigPair().signature().kind();
             final var preparer =
                     switch (kind) {
-                        case ECDSA_SECP256K1 -> ecPreparer == null
-                                ? ecPreparer = createPreparerForEC(signedBytes)
-                                : ecPreparer;
-                        case ED25519 -> edPreparer == null ? edPreparer = createPreparerForED(signedBytes) : edPreparer;
+                        case ECDSA_SECP256K1 -> ecPreparer;
+                        case ED25519 -> edPreparer;
                         case CONTRACT, ECDSA_384, RSA_3072, UNSET -> throw new IllegalArgumentException(
                                 "Unsupported signature type: " + kind);
                     };
 
+            assert preparer != null;
```

Comment:
Other places have been replaced with a thrown exception. Any reason this one isn't?

Reply:
I have added it because without it Sonar complains that _"A "NullPointerException" could be thrown; "preparer" is nullable here"_ which looking at the code does not seem possible to me and I didn't want to add a check that will be redundant.
Also for all other places with `assert`s Sonar was complaining but here it is fine with it.

Reply:
There's already a redundant check: `assert preparer != null`. Might as well make it look like the rest of the assertions imo. I'd prefer a regular throws statement, but I'm fine either way

Reply:
Updated to use use a regular runtime check as agreed with @rbair23 that the general approach will be such that  unless there is hard evidence, we will use normal runtime checks instead of asserts.